README.md

```diff
@@ -1,59 +1,49 @@
-[![rss on Crates.io](https://meritbadge.herokuapp.com/rss)](https://crates.io/crates/rss)
+[![Build Status](https://travis-ci.org/jameshurst/rss-rs.svg?branch=master)](https://travis-ci.org/jameshurst/rss-rs)
+[![Crates.io Status](http://meritbadge.herokuapp.com/rss-rs)](https://crates.io/crates/rss-rs)
+[![License](https://img.shields.io/badge/license-MIT-blue.svg)](https://raw.githubusercontent.com/jameshurst/rss-rs/master/LICENSE)
 
-[Documentation](https://frewsxcv.github.io/rust-rss/)
+[Documentation](https://jameshurst.github.io/rss-rs/rss/)
 
-Library for serializing the RSS web content syndication format.
+A fast RSS feed parser written in Rust. 
 
-## Examples
+rss-rs provides a complete implementation of the RSS 2.0 specification.
 
-### Writing
+## Usage
+
+To use rss-rs just add the dependency to your `Cargo.toml`.
+
+```toml
+[dependencies]
+rss-rs = "0.3"
+```
+
+The package includes a single crate named `rss`.
 
 ```rust
-use rss::{Channel, Item, Rss};
-
-let item = Item {
-    title: Some(String::from("Ford hires Elon Musk as CEO")),
-    pub_date: Some(String::from("01 Apr 2019 07:30:00 GMT")),
-    description: Some(String::from("In an unprecedented move, Ford hires Elon Musk.")),
-    ..Default::default()
-};
-
-let channel = Channel {
-    title: String::from("TechCrunch"),
-    link: String::from("http://techcrunch.com"),
-    description: String::from("The latest technology news and information on startups"),
-    items: vec![item],
-    ..Default::default()
-};
-
-let rss = Rss(channel);
-
-let rss_string = rss.to_string();
+extern crate rss;
 ```
 
-### Reading
+## Reading
+
+Reading can be done using any object that implements the `BufRead` trait. 
 
 ```rust
-use rss::Rss;
-
-let rss_str = r#"
-<?xml version="1.0" encoding="UTF-8"?>
-<rss version="2.0">
-  <channel>
-    <title>TechCrunch</title>
-    <link>http://techcrunch.com</link>
-    <description>The latest technology news and information on startups</description>
-    <item>
-      <title>Ford hires Elon Musk as CEO</title>
-      <pubDate>01 Apr 2019 07:30:00 GMT</pubDate>
-      <description>In an unprecedented move, Ford hires Elon Musk.</description>
-    </item>
-  </channel>
-</rss>
-"#;
-
-let rss = rss_str.parse::<Rss>().unwrap();
+let reader: BufRead = ...;
+let channel = Channel::read_from(reader).unwrap();
 ```
+
+## Extensions
+
+Elements which have non-default namespaces will be considered extensions. Extensions are stored in `Channel.extensions` and `Item.extensions`. 
+
+For conveninence, [Dublin Core](http://dublincore.org/documents/dces/) and [iTunes](https://help.apple.com/itc/podcasts_connect/#/itcb54353390) extensions are extracted to structs and stored in `Channel.itunes_ext`, `Channel.dublin_core_ext`, `Item.itunes_ext`, and `Item.dublin_core_ext`.
+
+## Invalid Feeds
+
+As a best effort to parse invalid feeds rss-rs will default elements declared as "required" by the RSS 2.0 specification to an empty string.
+
+## Todo
+
+* Writing support
```

Comment:
Writing support is definitely something I want to get in. If we were to merge this now we should make sure that writing is implemented before the next crates.io release.